components/script/dom/webglrenderingcontext.rs

```diff
@@ -167,31 +167,16 @@ impl WebGLRenderingContext {
     }
 
     #[allow(unsafe_code)]
-    fn float32_array_to_slice(&self, data: *mut JSObject) -> Option<Vec<f32>> {
+    fn array_buffer_view_to_vec<T: Clone>(&self, data: *mut JSObject) -> Option<Vec<T>> {
```

Comment:
While you’re at it, is the `.to_vec()` copy necessary? Or could this return `Option<&[T]>`? How long is the pointer from `JS_GetObjectAsArrayBufferView` valid?